src/components/withIntl.js

```diff
@@ -0,0 +1,66 @@
+import React, {Component} from 'react';
+import invariant from 'invariant';
+import createContext from 'create-react-context';
+import {invariantIntlContext} from '../utils';
+
+function getDisplayName(Component) {
+  return Component.displayName || Component.name || 'Component';
+}
+
+const IntlContext = createContext(null);
```

Comment:
Why don't you export the created IntlContext? It would be easier to use react-intl with the new react hooks RFC out of the box.

`
import { IntlContext } from 'react-intl';

export default () => {
  const intl = useContext(IntlContext);
  ...

}
`

Reply:
Simply because this pull request is not meant to add new functionality but instead to just upgrade to the new context API. Changes like this should be discussed in a separate issue, as directly exposing the context would expose `react-intl`'s internals :)

Reply:
If this work is going to result in a major version bump, can we re-evaluate exporting the IntlContext? Having access to the IntlContext would be very helpful for consumption with hooks and `contextType`.

Reply:
Agreed, exporting the IntlContext would be very beneficial for my team's usage.

Reply:
Or just export `useIntl = () => useContext(IntlContext)`

Reply:
Yes, in case a major bump gets agreed upon I think we can expose it ofc :)

Reply:
@danilotorrisi @tbiethman @phaux @bjankord Done in https://github.com/yahoo/react-intl/pull/1186/commits/5b52c620218296c8c76fd00a1dd664ef8686505c 👍